### PR TITLE
add more ut for kubelet_getters.go

### PR DESCRIPTION
**Pull request by njuptlzf**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
From
`k8s.io/kubernetes/pkg/kubelet/kubelet_getters.go (67.4%)`
To
`k8s.io/kubernetes/pkg/kubelet/kubelet_getters.go (86.8%)`

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by matthyx** [comment]
yay code coverage for kubelet, look at that @ehashman !

**Comment by ehashman** [comment]
flake #102605
/retest
/priority backlog
/triage accepted

/assign

File: pkg/kubelet/kubelet_getters_test.go
Comment (ehashman) [inline pkg/kubelet/kubelet_getters_test.go:183]: This doesn't create static pods, but we have logic that only triggers on static pod. It would be good to include a test case with static pods.

File: pkg/kubelet/kubelet_getters_test.go
Comment (ehashman) [inline pkg/kubelet/kubelet_getters_test.go:122]: Nit: maybe do this as an assertion?

File: pkg/kubelet/kubelet_getters_test.go
Comment (ehashman) [inline pkg/kubelet/kubelet_getters_test.go:178]: Do we have something that will invoke the error path of https://github.com/kubernetes/kubernetes/blob/8eff96dd12b5bde0fdf2d5d165c9d3575bc63d99/pkg/kubelet/kubelet_getters.go#L233-L239 to hit the "cannot get node" branch?

File: pkg/kubelet/kubelet_getters_test.go
Comment (ehashman) [inline pkg/kubelet/kubelet_getters_test.go:206]: Is `GetRequestedContainersInfo` testable? It's also uncovered.

**Review by ehashman — COMMENTED** [review]
Just some small areas for expansion while we're in here.

File: pkg/kubelet/kubelet_getters_test.go
Comment (njuptlzf) [reply pkg/kubelet/kubelet_getters_test.go:219]: Done. Improve coverage, but it doesn’t make sense for the time being.
There are many examples of fake returning `zero values`, which require additional optimization, but not here. :-)

File: pkg/kubelet/kubelet_getters_test.go
Comment (njuptlzf) [reply pkg/kubelet/kubelet_getters_test.go]: Done( actually covered by `ErrorAs`), modify the description information to make it clearer.

File: pkg/kubelet/kubelet_getters_test.go
Comment (njuptlzf) [reply pkg/kubelet/kubelet_getters_test.go]: Done

File: pkg/kubelet/kubelet_getters_test.go
Comment (njuptlzf) [reply pkg/kubelet/kubelet_getters_test.go]: Well, the two lines after this line construct pod[0] as a static pod, in order to make `GetPods` cover wider.
In addition, `but we have logic that only triggers on static pod` 
refers to the isStaticPod or isMirrorPod in pod_update.go?

 I did not find `getter.go` has logic similar to GetStaticPods, can you specify which function needs to be tested? :-)

Comment (ehashman) [reply pkg/kubelet/kubelet_getters_test.go]: I missed that, this is fine.

**Review by ehashman — COMMENTED** [review]
/skip
/lgtm

**Comment by njuptlzf** [comment]
This seems to require approver's permission, @mrunalp

**Comment by njuptlzf** [comment]
em... 
Looks like it needs to be approved again? I don’t know what happened, @mrunalp

**Review by mrunalp — APPROVED** [review]

**Comment by ehashman** [comment]
@njuptlzf please update your PR to fix the verify issue!

**Comment by dims** [comment]
this PR has been on hold for about 2 weeks. Can you please see how to unblock this by reviewing comments etc? or please close this out (Trying to declutter our open list of PRs)

**Comment by ehashman** [comment]
/lgtm cancel

since unmergable in current state

**Pull request closed without merging**
